provider/grpc.go

```diff
@@ -222,43 +226,52 @@ func (g *GRPCClient) SubmitTradeSwap(ctx context.Context, ownerAddress, inToken,
 		return nil, err
 	}
 
-	var signatures []string
+	batchRequest := pb.PostSubmitBatchRequest{}
+	batchRequest.SubmitStrategy = submitStrategy
 	for _, tx := range resp.Transactions {
-		signature, err := g.signAndSubmit(ctx, tx, skipPreFlight)
+		oneRequest := pb.PostSubmitRequestEntry{}
+		oneRequest.SkipPreFlight = skipPreFlight
+		signedTxBase64, err := transaction.SignTxWithPrivateKey(tx.Content, *g.privateKey)
 		if err != nil {
-			if signature != "" {
-				signatures = append(signatures, signature)
-			}
-			return signatures, err
+			return nil, err
 		}
-
-		signatures = append(signatures, signature)
+		oneRequest.Transaction = &pb.TransactionMessage{
+			Content:   signedTxBase64,
+			IsCleanup: tx.IsCleanup,
+		}
+		batchRequest.Entries = append(batchRequest.Entries, &oneRequest)
 	}
 
-	return signatures, nil
+	return g.PostSubmitBatch(ctx, &batchRequest)
 }
 
 // SubmitRouteTradeSwap builds a RouteTradeSwap transaction then signs it, and submits to the network.
-func (g *GRPCClient) SubmitRouteTradeSwap(ctx context.Context, request *pb.RouteTradeSwapRequest, skipPreFlight bool) ([]string, error) {
-	resp, err := g.apiClient.PostRouteTradeSwap(ctx, request)
+func (g *GRPCClient) SubmitRouteTradeSwap(ctx context.Context, request *pb.RouteTradeSwapRequest, submitStrategy pb.SubmitStrategy, skipPreFlight bool) (*pb.PostSubmitBatchResponse, error) {
+	if g.privateKey == nil {
+		return nil, ErrPrivateKeyNotFound
+	}
+
+	resp, err := g.PostRouteTradeSwap(ctx, request)
 	if err != nil {
 		return nil, err
 	}
-
-	var signatures []string
+	batchRequest := pb.PostSubmitBatchRequest{}
```

Comment:
this whole section can be refactor into a single function to be shared across the other submission endpoints, including the `g.privateKey == nil` check